core/src/test/scala/caliban/validation/ValidationSchemaSpec.scala

```diff
@@ -288,9 +288,45 @@ object ValidationSchemaSpec extends DefaultRunnableSpec {
               "Field 'a' in Object 'ObjectWrongListItemSubtype' is an invalid list item subtype"
             )
           },
+          /* I think this test is broken, I can't get it to generate anything other than
+
+          schema {
+            query: TestNonNullableObject
+          }
+
+          interface WithNullable {
+            field: String!
+          }
+
+          type IsNonNullable implements WithNullable {
+            field: String!
+          }
+
+          type IsNullable implements WithNullable {
+            field: String
+          }
+
+          type TestNonNullableObject {
+            nonNullable: WithNullable!
+          }
+
+          which obviously isn't testing what we want.
+
+          The previous test generated an empty interface since
+          it only had a single case class so no field intersection
+          could be found.
+
+          Not sure how to make it so that the interface becomes
```

Comment:
In `SchemaDerivation` (under scala-2, but similar code in scala-3) line 131, we group fields by name and if they have the same type then take `headOption` (which is an arbitrary one). I guess we need some extra logic to take the optional one instead of an arbitrary one.

Reply:
Great, I'll do some digging!

Reply:
> Doesn't it create an interface even if there is only one extending class?

The reason for that was probably `TestNonNullableObject` returning `WithNonNullable` rather than `WithNullable` now that I think of it.

Reply:
I fixed this in cc57370, WDYT of that way of doing it instead?

Reply:
Oh, totally missed Scala 3 🤦

Reply:
Fixed in 70c6302.